File: app/src/main/java/com/suhang/networkmvp/mvp/model/AttentionModel.java
package com.suhang.networkmvp.mvp.model;

import android.util.ArrayMap;

import com.suhang.networkmvp.domain.AppMain;
import com.suhang.networkmvp.domain.GithubBean;
import com.suhang.networkmvp.interfaces.INetworkOtherService;
import com.suhang.networkmvp.interfaces.INetworkService;
import com.suhang.networkmvp.mvp.IModel;

import javax.inject.Inject;

/**
 * Created by 苏杭 on 2017/4/28 16:33.
 */

public class AttentionModel extends BaseModel implements IAttentionModel{
    public static final int APPMAIN = 1;
    public static final int GITHUB = 2;
    public static final int TAG_APP = 3;
    public static final int TAG_GIT = 4;
    @Inject
    NetworkModel mModel;

    @Inject
    public AttentionModel() {
    }


    public void getAppMainData() {
        mModel.loadPostDataWrap(AppMain.class,  false, TAG_APP,new ArrayMap<>());
    }

    public void getGithubData() {
        mModel.loadGetData(GithubBean.class, TAG_GIT,"2/1");
    }

    @Override
    public AppMain getData() {
        return null;
    }

    @Override
    public void getAppMain() {

    }
}

File: app/src/main/java/com/suhang/networkmvp/mvp/model/AttentionModel.kt
package com.suhang.networkmvp.mvp.model

import android.util.ArrayMap
import com.suhang.networkmvp.constants.URLS
import com.suhang.networkmvp.interfaces.INetworkManager
import org.jetbrains.anko.info
import javax.inject.Inject

/**
 * Created by 苏杭 on 2017/4/28 16:33.
 */
class AttentionModel @Inject constructor():IAttentionModel {
    override fun destroy() {
        info(mManager)
    }

    @Inject
    lateinit var mManager: INetworkManager



    override fun getAppMainData() {
        val flow1 = mManager.getFlowableWrap(URLS.URL_APPMAIN,whichTag = 111,params = ArrayMap<Any,Any>())
        val flow3 = mManager.getFlowableWrap(URLS.URL_APPMAIN,whichTag = 333,params = ArrayMap<Any,Any>())
        val flow2 = mManager.getFlowable(URLS.URL_GITHUB,whichTag = 222, params = "2/1")
        mManager.zip(flow1,flow2,flow3)
//        mManager.loadPostDataWrap(URLS.URL_APPMAIN,needCache = true,params = ArrayMap<Any,Any>())
    }

    override fun getGithubData() {
        mManager.loadGetData(URLS.URL_GITHUB, params = "2/1")
    }
}
